lib/api_callable.js

```diff
@@ -391,44 +292,38 @@ Bundleable.prototype.call = function(apiCall, argument, settings) {
  *   a bare rpc call. This is a Promise instead of a bare function because
  *   the rpc call will be involeved with asynchronous authentications.
  * @param {CallSettings} settings - provides the settings for this call
+ * @param {Object=} optDescriptor - optionally specify the descriptor for
+ *   the method call.
  * @return {APICall} func - a bound method on a request stub used
  *   to make an rpc call.
  * @throws - if `settings` has incompatible values, e.g, if bundling
  *   and page_streaming are both configured.
  */
-exports.createApiCall = function createApiCall(funcWithAuth, settings) {
-  if (settings.pageDescriptor && settings.bundler) {
-    throw new Error('The API call has incompatible settings: ' +
-        'bundling and page streaming');
-  }
+exports.createApiCall = function createApiCall(
+    funcWithAuth, settings, optDescriptor) {
+  var apiCaller =
+      optDescriptor ? optDescriptor.apiCaller(settings) : new NormalApiCaller();
 
   return function apiCallInner(request, callOptions, callback) {
     var thisSettings = settings.merge(callOptions);
-    var apiCaller;
-    if (thisSettings.pageDescriptor) {
-      apiCaller = new PageStreamable(thisSettings.pageDescriptor);
-    } else if (thisSettings.bundler) {
-      apiCaller = new Bundleable(thisSettings.bundler);
-    } else {
-      apiCaller = new NormalApiCaller();
-    }
 
-    var result = apiCaller.init(thisSettings, callback);
+    var status = apiCaller.init(thisSettings, callback);
```

Comment:
(I think I've replied here, but somehow not seen right now, so saying again...)

I'm open to renaming this.
The justification for changing from "result" is, it's not a 'result' anymore, the result can be extracted from a method below. So now it handles a "status" of current method invocation, that's why I changed to "status". But I'm not thinking this is the best name. Please let me know if you find out a better name.